pkg/k8shandler/util.go

```diff
@@ -313,3 +314,66 @@ func popStatefulSet(statefulSets *apps.StatefulSetList, cfg desiredNodeState) (*
 	dpls.Items = statefulSets.Items[:len(statefulSets.Items)-1]
 	return dpls, statefulSet, true
 }
+
+func getMasterCount(dpl *v1alpha1.Elasticsearch) int32 {
+	masterCount := int32(0)
+
+	for _, node := range dpl.Spec.Nodes {
+		if isNodeMaster(node) {
+			masterCount = masterCount + node.Replicas
+		}
+	}
+
+	return masterCount
+}
+
+func getDataCount(dpl *v1alpha1.Elasticsearch) int32 {
+	dataCount := int32(0)
+
+	for _, node := range dpl.Spec.Nodes {
+		if isNodeData(node) {
+			dataCount = dataCount + node.Replicas
+		}
+	}
+
+	return dataCount
+}
+
+func isNodeMaster(node v1alpha1.ElasticsearchNode) bool {
+	for _, role := range node.Roles {
+		if role == v1alpha1.ElasticsearchRoleMaster {
+			return true
+		}
+	}
+	return false
+}
+
+func isNodePureMaster(node v1alpha1.ElasticsearchNode) bool {
+	if len(node.Roles) > 1 {
```

Comment:
It's just a list, so technically it could have duplicate values. I'm not sure if this function really has use, but I haven't looked through the code enough to see.